src/Symfony/Bridge/Doctrine/PropertyInfo/DoctrineExtractor.php

```diff
@@ -117,6 +117,15 @@ public function getTypes($class, $property, array $context = array())
                 case DBALType::TIME:
                     return array(new Type(Type::BUILTIN_TYPE_OBJECT, $nullable, 'DateTime'));
 
+                case 'date_immutable':
+                case 'datetime_immutable':
+                case 'datetimetz_immutable':
```

Comment:
Only the `time` type was tested (and not the `date`, `datetime` and `datetimetz`) so I decided to do the same for the immutable variants.